crates/hir/src/semantics.rs

```diff
@@ -1458,6 +1470,59 @@ impl<'db> SemanticsImpl<'db> {
             .map(|ty| ty.original.is_packed(self.db))
             .unwrap_or(false)
     }
+
+    fn is_inside_unsafe(&self, node: &SyntaxNode) -> bool {
```

Comment:
```suggestion
    fn is_inside_unsafe(&self, node: &ast::Expr)-> bool {
```
This handles expressions so it should only take such